admin-dev/themes/default/js/bundle/product/form.js

```diff
@@ -1771,57 +1793,296 @@ var priceCalculation = (function() {
+      this.impactFinalPriceTaxIncluded(obj);
+    },
 
-      finalPrice.html(priceToBeChanged);
+    /**
+     * @param {jQuery} obj
+     */
+    impactFinalPriceTaxExcluded(obj) {
+      var combinationForm = obj.closest('div[id^="combination_form_"]');
+
+      var impactPriceTE = this.getImpactTEInputValue(obj);
+      var ecotaxTE = this.getCombinationEcotaxTaxExcluded(obj);
+      // If no ecotax for combination use the product's one
+      if (ecotaxTE <= 0) {
+        ecotaxTE = getEcotaxTaxExcluded();
+      }
+
+      var finalPriceContainer = combinationForm.find('.final-price');
+      var productPriceTE = priceCalculation.getProductBasePrice();
+      var finalPriceTE = productPriceTE + impactPriceTE + ecotaxTE;
+      finalPriceTE = ps_round(finalPriceTE, displayPricePrecision);
+      finalPriceTE = truncateDecimals(finalPriceTE, displayPricePrecision);
+
+      finalPriceContainer.html(finalPriceTE);
+      finalPriceContainer.data('price', finalPriceTE);
+    },
+
+    /**
+     * @param {jQuery} obj
+     */
+    impactFinalPriceTaxIncluded(obj) {
+      var combinationForm = obj.closest('div[id^="combination_form_"]');
+
+      var impactPriceTE = this.getImpactTEInputValue(obj);
+      var ecotaxTI = this.getCombinationEcotaxTaxIncluded(obj);
+      // If no ecotax for combination use the product's one
+      if (ecotaxTI <= 0) {
+        ecotaxTI = getEcotaxTaxIncluded();
+      }
+
+      var finalPriceTIContainer = combinationForm.find('.final-price-tax-included');
+      var productPriceTE = priceCalculation.getProductBasePrice();
+      var finalPriceTI = this.computePriceTaxIncluded(productPriceTE + impactPriceTE) + ecotaxTI;
+      finalPriceTI = ps_round(finalPriceTI, displayPricePrecision);
+      finalPriceTI = truncateDecimals(finalPriceTI, displayPricePrecision);
+
+      finalPriceTIContainer.html(finalPriceTI);
+      finalPriceTIContainer.data('price', finalPriceTI);
+    },
+
+    /**
+     * Calculates the impact on price so that the change on ecotax doesn't affect the final price
+     * @param {jQuery} obj
+     */
+    impactPricesForEcotax: function (obj) {
+      var finalPriceTIContainer = obj.closest('div[id^="combination_form_"]').find('.final-price-tax-included');
+      var currentFinalPriceTI = Number(finalPriceTIContainer.data('price'));
+      var productPrice = priceCalculation.getProductBasePrice();
+      var productPriceTI = this.computePriceTaxIncluded(productPrice);
+      var ecotaxTI = this.getCombinationEcotaxTaxIncluded(obj);
+      // If no ecotax for combination use the product's one
+      if (ecotaxTI <= 0) {
+        ecotaxTI = getEcotaxTaxIncluded();
+      }
+
+      // Compute impact price tax excluded then update the price tax included
+      var impactPriceTI = currentFinalPriceTI - ecotaxTI - productPriceTI;
+      impactPriceTI = ps_round(impactPriceTI, displayPricePrecision);
+      impactPriceTI = truncateDecimals(impactPriceTI, displayPricePrecision);
```

Comment:
So we truncate after round. I guess there will be some usecases where this will give us inaccurate computations 😞

Reply:
Honestly, I'm not sure the truncate is really useful, since we already round with the precision before.. But I copied some existing code from the current product page to remain consistent

Reply:
I understand 👍